WalletWasabi/Blockchain/BlockFilters/IndexBuilderService.cs

```diff
@@ -118,148 +90,105 @@ public void Synchronize()
 			{
 				try
 				{
-					if (Interlocked.Read(ref _runner) >= 2)
+					if (Interlocked.Read(ref _workerCount) >= 2)
 					{
 						return;
 					}
 
-					Interlocked.Increment(ref _runner);
-					while (Interlocked.Read(ref _runner) != 1)
+					Interlocked.Increment(ref _workerCount);
+					while (Interlocked.Read(ref _workerCount) != 1)
 					{
 						await Task.Delay(100);
 					}
 
-					if (Interlocked.Read(ref _running) >= 2)
+					if (Interlocked.Read(ref _serviceStatus) >= Stopping)
 					{
 						return;
 					}
 
 					try
 					{
-						Interlocked.Exchange(ref _running, 1);
+						Interlocked.Exchange(ref _serviceStatus, Running);
 
-						var isImmature = false; // The last 100 blocks are reorgable. (Assume it is mature at first.)
 						SyncInfo syncInfo = null;
 						while (IsRunning)
 						{
 							try
 							{
 								// If we did not yet initialized syncInfo, do so.
-								if (syncInfo is null)
+								if (syncInfo == null)
```

Comment:
https://github.com/zkSNACKs/WalletWasabi/blob/master/CONTRIBUTING.md#null-check
```suggestion
								if (syncInfo is null)
```